### DISCUSSION TITLE
Issues when using `pyright` + `multimethod`

### DISCUSSION THREAD

**Discussion by bzm3r** [microsoft/pyright#8254] [Q&A] [answered]
Consider the following little example:

```
from multimethod import multimethod


@multimethod
def normalize_index(index, labels: list[int]) -> int | slice | list[int]:
    raise Exception(f"No logic to normalize {index=} of {type(ix)=}")


@normalize_index.register(int | slice, list[int])
def _(index: int | slice, _: list[int]) -> int | slice | list[int]:
    return index
```

It produces the following errors for me, both on the line containing `@normalize_index.register(int | slice, list[int])`:

1. An issue with `register`, imported from `multimethod`:

```
Cannot access attribute "register" for class "_Wrapped[..., Unknown, ..., Unknown]"
  Attribute "register" is unknown
```

2. An issue with using `UnionType` for `type`:  

```
Argument of type "UnionType" cannot be assigned to parameter "args" of type "type" in function "register"
  "UnionType" is incompatible with "type"
```
Is there anything I can do from my end to help these errors go away?

**Comment by erictraut** [comment, marked as answer]
I'm not familiar with this library. It looks like it contains some inline type annotations, but it's very incomplete. You could try creating your own stub file that is more complete. Short of that, there's probably not much you can do here to eliminate the static type errors.

**Reply by bzm3r** [reply]
Makes sense! I will address the errors then by simply not using it, as I don't have the time to contribute detailed type information post-hoc.
